statsmodels/distributions/empirical_distribution.py

```diff
@@ -144,6 +144,54 @@ def __init__(self, x, side='right'):
         # which would have raised a NameError if hit, so would need to be
         # fixed.  See GH#5701.
 
+class ECDFDiscrete(StepFunction):
+    """
+    Return the Empirical Weighted CDF of an array as a step function.
+
+    Parameters
+    ----------
+    x : array_like
+        Observations
```

Comment:
change to:

Data values. If freq_weights is None, then x is treated as observations and the ecdf is computed from the frequency counts of unique values using `nunpy.unique`. If freq_weights is not None, then x will be taken as the support of the mass point distribution with freq_weights as counts for x values.
The x values can be arbitrary sortable values and need not be integers.